src/vs/platform/backup/electron-main/backupMainService.ts

```diff
@@ -23,7 +28,9 @@ export class BackupMainService implements IBackupMainService {
 	protected backupHome: string;
 	protected workspacesJsonPath: string;
 
-	protected backups: IBackupWorkspacesFormat;
+	protected rootWorkspaces: IWorkspaceIdentifier[];
```

Comment:
Didn't `backups` encapsulate these 3 properties before inside `IBackupWorkspacesFormat`? Why are they being broken out?

Also if it's not clear, the backups are stored on an entire workspace level; you can have multiple backups for a single folder if they're in different folders

Reply:
@aeschli Can you please address @Tyriar comments?

Reply:
The `IBackupWorkspacesFormat` is the format of the persisted data on disk. With the change is got an additional field `folderURIWorspaces` also `folderWorkspaces` is now depreacted . All these fields are using `string[]`. In the loaded model we want to keep `URL[]` for folderWorkspaces. Given that, keeping the model as `IBackupWorkspacesFormat` was no longer practical. I added separate fields `rootWorkspaces`, `folderWorkspaces` (URI[]), and `emptyWorkspaces` to class `BackupWorkspacesFormat`.

Sorry, I know this isn't a minimal change, but while at it, I thought it's better to untangle some of the code. We'll touch this code again when adding support workspaces loaded from URIs.